tools/Vis_PV.py

```diff
@@ -0,0 +1,356 @@
+    variable_lookup_table = GetColorTransferFunction(var)
+    variable_lookup_table.ApplyPreset(color_map, True)
+    # Show the color bar for the variable being visualized:
+    xdmf_reader_display.SetScalarBarVisibility(render_view, True)
+    xdmf_reader_display.SetScaleArray = ['POINTS', var]
+    xdmf_reader_display.ScaleTransferFunction = 'PiecewiseFunction'
+    return xdmf_reader_display, render_view, variable_lookup_table
+
+
+def tetrahedralize(xdmf_reader, render_view):
+    '''
+    Apply tetrahedralize filter and update view
+    '''
+    tetrahedralize1 = Tetrahedralize(Input=xdmf_reader)
+    Hide(xdmf_reader, render_view)
+    xdmf_reader_display = Show(tetrahedralize1, render_view)
+    render_view.Update()
+    return render_view, xdmf_reader_display
+
+
+def set_opacity(var, function_type, opacity_val, render_view,\
+                xdmf_reader, var_lut):
+    '''
+    Set opacity based on option chosen in yaml input file:
+    'Constant': some value between 0 and 1
+    'Proportional': opacity set proportional to the variable being visualized,
+    (Under 'Variable_name' in 'Variable_properties')
+    '''
+    if function_type != 'Constant' and function_type != 'Proportional':
+        sys.exit("The Opacity option specified in the input file is invalid.\
+        Enter either 'Constant' or 'Proportional' for the function type")
+    # Get range of var array
+    var_range = xdmf_reader.PointData.GetArray(var).GetRange()
+    # Set variables to construct gaussians
+    var_max = var_range[1]
+    var_min = var_range[0]
+
+    if function_type == 'Constant':  # Constant Opacity
+        var_lut.EnableOpacityMapping = 1
+        Var_PWF = GetOpacityTransferFunction(var)
+        Var_PWF.Points = [var_min, opacity_val, 0.5,
+                          0.0, var_max, opacity_val, 0.5, 0.0]
+
+    elif function_type == 'Proportional':  # Varying opacity
+        var_lut.EnableOpacityMapping = 1
+        Var_PWF = GetOpacityTransferFunction(var)
+        num_points = 200  # Number of points to evaluate opacity function
+        num_gauss = 5  # Number of points to evaluate gaussians
+        var_values = np.asarray(np.linspace(
+            var_min, var_max, num_points))  # Array of var values
+        center_values = np.asarray(np.linspace(
+            var_min, var_max, num_gauss))  # centersof gaussian
+        # amplitudes of gaussians
+        amplitude_values = abs(center_values)/max(abs(var_max), abs(var_min))
+        sigma = (var_values[1] - var_values[0]) * 2.0
+
+        gaussians = []
+        gaussians = [np.asarray([center_values[i], amplitude_values[i]])
+                     for i in range(num_gauss)]
+        opacity_function = np.zeros(len(var_values))
+        for gaussian in gaussians:
+            opacity_function += gaussian[1] * np.exp(-1.0 * np.square(
+                var_values - gaussian[0])/(2.0 * math.pow(sigma, 2)))
+        # Create opacity list with var_values, opacity function and 0.0, 0.5
```

Comment:
Could you please expand this comment? I don't understand what this is actually doing. Why aren't the Guassians enough?

Reply:
ping